ci/local-run.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail -u
+
+function check_var() {
+  [[ ${!1-} ]] && return 0 || true
+  [[ -f .env ]] || { echo "Var $1 missing from environment and no .env file found; see .env.sample file"; return 1; }
+  grep -q "^$1=" .env || { echo "Var $1 missing both from environment and .env file"; return 1; }
+}
+
+image_name=$1
+
+if [[ $OSTYPE == 'darwin'* ]]; then  # group id 20 is usually taken in MacOS
```

Comment:
Hmmm what's the specific problem that happened with group 20? It shouldn't be a problem if the group is _taken_ on the host system... in fact we want to specify `group_id` build ARG that's the indeed the same as the group id in the host system, in the spirit of https://medium.com/virtuslab/nifty-docker-tricks-for-your-ci-vol-2-c5191a67f1a4 -> Ctrl+F `if only`

Reply:
<img width="277" alt="Screenshot 2021-08-25 at 07 37 54" src="https://user-images.githubusercontent.com/73847352/130740060-ef93e469-5e03-4dd4-ac49-e44b1a1e9957.png">
AFAICS this is quite common on mac..

Reply:
Hmmm okay but this is failing within the Docker image, so the error is caused by group id 20 being taken in the **base Docker image** and not in Mac OS ;)

Reply:
I think the easiest fix here would be to first check in Dockerfile whether the group to be created exists, and skip the creation if it already does :)

Reply:
I resigned from implementing this.  Check if group exist can be done easily but later on, while calling `adduser` i need to pass exact name of group. This is because `adduser` in alpine comes from busybox and `--gid` option is not recognized. On the other hand i cannot also retrieve exact name of group with `grep -w $group_id /etc/group | cut -d: -f4`, because if group was created in earlier step, this info is not yet available.

Reply:
This is the solution i could find and wo
![Uploading Screenshot 2021-08-25 at 15.11.05.png…]()
rk in every case but it's far from being pretty: